packages/components/menu/src/menu.ts

```diff
@@ -347,37 +284,71 @@ export default defineComponent({
       }
     })
 
-    return {
+    expose({
+      open,
+      close,
       hoverBackground,
-      isMenuPopup,
-      menu,
-      filteredSlot,
+    })
 
-      props,
+    const useVNodeResize = (vnode: VNode) => props.mode === 'horizontal' ? withDirectives(vnode, [[Resize, handleResize]]) : vnode
+    return () => {
 
-      open,
-      close,
-      handleResize,
-    }
-  },
-  render() {
-    const directives = this.mode === 'horizontal' ? [[Resize, this.handleResize]] : []
-    const menu = withDirectives(h('ul', {
-      key: String(this.collapse),
-      role: 'menubar',
-      ref: 'menu',
-      style: { backgroundColor: this.backgroundColor || '' },
-      class: {
-        'el-menu': true,
-        'el-menu--horizontal': this.mode === 'horizontal',
-        'el-menu--collapse': this.collapse,
-      },
-    }, [this.filteredSlot]), directives)
+      let slot = slots.default?.() ?? []
+      const showMore = []
+
+      if (props.mode === 'horizontal') {
+        const items = Array.from((menu.value as Node | undefined)?.childNodes ?? []).filter(item => item.nodeName !== '#text' || item.nodeValue) as HTMLElement[]
+        const originalSlot = slot.flat(Infinity)
+        if (items.length === originalSlot.length) {
+          const moreItemWidth = 64
+          const paddingLeft = parseInt(getComputedStyle(menu.value).paddingLeft, 10)
+          const paddingRight = parseInt(getComputedStyle(menu.value).paddingRight, 10)
+          const menuWidth = menu.value.clientWidth - paddingLeft - paddingRight
+          let calcWidth = 0
+          let sliceIndex = 0
+          items.forEach((item, index) => {
+            calcWidth += item.offsetWidth || 0
+            if (calcWidth <= menuWidth - moreItemWidth) {
+              sliceIndex = index + 1
+            }
+          })
+          const defaultSlot = originalSlot.slice(0, sliceIndex)
+          const moreSlot = originalSlot.slice(sliceIndex)
+          if (moreSlot?.length) {
+            slot = defaultSlot
+            showMore.push(h(ElSubMenu, {
+              index: 'sub-menu-more',
+              class: 'el-sub-menu__hide-arrow',
+            }, {
+              title: () => h('i', { class: ['el-icon-more', 'el-sub-menu__icon-more'] }),
+              default: () => moreSlot,
+            }))
+          }
+        } else {
+          nextTick(() => instance.proxy.$forceUpdate())
```

Comment:
Why not
```ts
nextTick(instance.proxy.$forceUpdate)
```

Reply:
Same. Prevent the parameters of `nextTick` from being inadvertently passed to `$forceUpdate`